dist/src/main/java/io/camunda/application/commons/backup/BackupPriorityConfiguration.java

```diff
@@ -102,37 +122,25 @@ private <A> Function<Map<String, A>, String> differentConfigFor(final String fie
 
   @Bean
   public BackupPriorities backupPriorities() {
-    final var indexPrefix =
-        allMatch(
-                NO_CONFIG_ERROR_MESSAGE,
-                differentConfigFor("indexPrefix"),
-                Map.of(
-                    "operate",
-                    Optional.ofNullable(operateProperties).map(OperateProperties::getIndexPrefix),
-                    "tasklist",
-                    Optional.ofNullable(tasklistProperties)
-                        .map(TasklistProperties::getIndexPrefix)),
-                skipEmptyOptional())
-            .get();
-    final var isElasticsearch =
-        allMatch(
-                NO_CONFIG_ERROR_MESSAGE,
-                differentConfigFor("database.type"),
-                Map.of(
-                    "operate",
-                    Optional.ofNullable(operateProperties)
-                        .map(ignored -> DatabaseInfo.isCurrent(DatabaseType.Elasticsearch)),
-                    "tasklist",
-                    Optional.ofNullable(tasklistProperties)
-                        .map(prop -> prop.getDatabase().equals(TasklistProperties.ELASTIC_SEARCH))),
-                skipEmptyOptional())
-            .get();
+    final var indexPrefix = getIndexPrefix();
+
+    final boolean isElasticsearch = getIsElasticsearch();
+
+    final var indexService =
+        new OptimizeIndexNameService(indexPrefix.isBlank() ? "optimize" : indexPrefix);
```

Comment:
I think the default case is handled already by Optimize config, and I _believe_ that an empty assignment would already default to `optimize`. If possible, would rather have the defaulting handled in one place

Reply:
Yes, that was one thing that I needed to double check with you, but I forgot to ask beforehand :+1:

Reply:
So when apps different from optimize are deployed standalone (i.e. without optimize) the prefix will still be equal to "optimize" right? the service-config.yaml is  included as well for those apps?

Reply:
I tried removing `required=false` from `@Autowired(required=false) OptimizeIndexNameService` but then the application is unable to start.
For this reason I added as a last resort `new OptimizeIndexNameProvider(defaultIndexPrefix)`